### PR TITLE
Allow option to display absolute values for pie chart #19338

### LINKED ISSUE #19338: Allow option to display absolute values for pie chart
### Problem

I would like to show a label displaying the absolute numbers used to create the pie chart. Currently, only percentage labels are supported via `autopct`. Hacky solutions like https://stackoverflow.com/a/48299798/3779655 exist to retrieve the absolute values based from the fractional ones. It would be better to support it out of the box.

### Proposed Solution

Add option `wedgelabels=[None|'percent','absolute']` together with `wedgelabelfmt` replacing `autopct`.

For migration, initialize `wedgelabelfmt` with `autopct` for `wedgelabels='percent'` and overwrite if `wedgelabelfmt` is set.

### PR DESCRIPTION
**Pull request by codebreakymefix**
## PR Summary
Adds a parameter 'wedgelabels' to pie() for the user to display absolute or percent values in the wedges. Renamed 'autopct' to "wedgelabelsfmt" as the user requested. 

closes #19338
## PR Checklist

**Linked Issue**
- [x] Added "closes #0000" in the PR description to link it to the original issue.

**Documentation and Tests**
- [x] Has pytest style unit tests (and `pytest` passes)
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -3069,9 +3069,9 @@ def stem(self, *args, linefmt=None, markerfmt=None, basefmt=None, bottom=0,
 
     @_preprocess_data(replace_names=["x", "explode", "labels", "colors"])
     def pie(self, x, explode=None, labels=None, colors=None,
-            autopct=None, pctdistance=0.6, shadow=False, labeldistance=1.1,
-            startangle=0, radius=1, counterclock=True,
-            wedgeprops=None, textprops=None, center=(0, 0),
+            autopct=None, wedgelabels=None, wedgelabelsfmt=None, pctdistance=0.6,
+            shadow=False, labeldistance=1.1, startangle=0, radius=1,
+            counterclock=True, wedgeprops=None, textprops=None, center=(0, 0),
             frame=False, rotatelabels=False, *, normalize=True, hatch=None):
```
Comment (oscargus) [inline lib/matplotlib/axes/_axes.py:3075]: The new arguments must be added after the `*` in case someone is using positional arguments.

**Review by oscargus — CHANGES_REQUESTED** [review]
Thanks for this!

It is not clear to me how this works in practice. Right now, both `autofmt` and `wedgelabelsfmt` does the same thing? If so, why should we have both? If not, how can we deprecate `autofmt` later?

If it is only a matter of a "bad" name, then I think it is better to just have one name and use the decorator `@_api.rename_parameter("3.8", "autofmt", "wedgelabelsfmt")`.

However, I would wait for additional input before going all in on these changes. (New arguments after `*` is a requirement though.)

Edit: after reading the issue, I think it should be quite OK to add the decorator as well.

**Comment by rcomer** [comment]
Hi @codebreakymefix are you still interested in working in this?  It looks like maybe you were waiting for reviewers to reach a consensus.

I think I agree with @story645 on this that the proposed _wedgelabels_ could be confusing.  As the [pie example](https://matplotlib.org/stable/gallery/pie_and_polar_charts/pie_features.html#swap-label-and-autopct-text-positions) shows, the current _labels_ can be put on the wedge and the automatic percentage labels outside.

Perhaps _autolabels_ and _autolabelfmt_ rather than _wedge…_ might work?  Or I also like @story645’s suggestion of an “absolute” format keyword that can’t be passed together with _autopct_.

**Comment by rcomer** [comment]
Discussion on parameter name also needs to consider the _pctdistance_ parameter.

**Pull request closed without merging**